pkg/volume/csi/csi_block.go

```diff
@@ -367,5 +367,10 @@ func (m *csiBlockMapper) TearDownDevice(globalMapPath, devicePath string) error
 		}
 	}
 
+	pvPath := filepath.Dir(globalMapPath)
+	err = os.RemoveAll(pvPath)
```

Comment:
I don't feel well when I see `os.RemoveAll`. In case of a driver failure, it *can* remove user data, right? Even if it is a symlink / bind mount to a real device.

`os.Remove` would be IMO enough here, reporting errors if it fails (so user knows the driver left some stale data there).

Reply:
Yeap, but usually data.vol.json would still exist, which would cause os.Remove() failed.
So, I think, we should remove the file first.

Reply:
@jsafrane wdyt?